src/Sarif.Viewer.VisualStudio/ErrorList/SarifErrorListEventProcessor.cs

```diff
@@ -0,0 +1,136 @@
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public event EventHandler<SarifErrorListSelectionChangedEventArgs> NavigatedItemChanged;
+        #endregion ISarifErrorListEventSelectionService
+
+        public IWpfTableControl errorListTableControl;
+
+        /// <summary>
+        /// Called by <see cref="SarifErrorListEventProcessorProvider"/> to set the table this service will
+        /// handle.
+        /// </summary>
+        /// <param name="wpfTableControl">The WPF table control representing the error list.</param>
+        public void SetTableControl(IWpfTableControl wpfTableControl)
+        {
+            this.errorListTableControl = wpfTableControl;
+        }
+
+        public override void PostprocessSelectionChanged(TableSelectionChangedEventArgs e)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            base.PostprocessSelectionChanged(e);
+
+            if (this.errorListTableControl == null)
+            {
+                return;
+            }
+
+            // Make sure there is only one selection, that's all we support.
+            IEnumerator<ITableEntryHandle> enumerator = (this.errorListTableControl.SelectedEntries ?? Enumerable.Empty<ITableEntryHandle>()).GetEnumerator();
+            ITableEntryHandle selectedTableEntry = null;
+            if (enumerator.MoveNext())
+            {
+                selectedTableEntry = enumerator.Current;
+
+                if (enumerator.MoveNext())
+                {
+                    selectedTableEntry = null;
+                }
+            }
+
+            SarifErrorListItem selectedSarifErrorItem = null;
+            if (selectedTableEntry != null)
+            {
+                TryGetSarifResult(selectedTableEntry, out selectedSarifErrorItem);
+            }
+
+            SarifErrorListItem previouslySelectedItem = this.currentlySelectedItem;
+            this.currentlySelectedItem = selectedSarifErrorItem;
+
+            SelectedItemChanged?.Invoke(this, new SarifErrorListSelectionChangedEventArgs(previouslySelectedItem, this.currentlySelectedItem));
+        }
+
+        public override void PostprocessNavigate(ITableEntryHandle entry, TableEntryNavigateEventArgs e)
+        {
+            base.PostprocessNavigate(entry, e);
+
+            this.TryGetSarifResult(entry, out SarifErrorListItem newlyNavigatedErrorItem);
+
+            SarifErrorListItem previouslyNavigatedItem = this.currentlyNavigatedItem;
+            this.currentlyNavigatedItem = newlyNavigatedErrorItem;
+
+            NavigatedItemChanged?.Invoke(this, new SarifErrorListSelectionChangedEventArgs(previouslyNavigatedItem, this.currentlyNavigatedItem));
+        }
+
+        private bool TryGetSarifResult(ITableEntryHandle entryHandle, out SarifErrorListItem sarifResult)
+        {
+            sarifResult = null;
+
+            if (entryHandle.TryGetEntry(out ITableEntry tableEntry) && tableEntry is SarifResultTableEntry sarifResultTableEntry)
```

Comment:
>tableEntry is SarifResultTableEntry sarifResultTableEntr [](start = 71, length = 56)

Add a comment that this prevents us from navigating to someone else's error item (such as a IntelliSense error)